include/ledstripeffect.h

```diff
@@ -48,9 +51,23 @@ class LEDStripEffect : public IJSONSerializable
     size_t _cLEDs;
     String _friendlyName;
     int    _effectNumber;
+    bool   _enabled = true;
+    std::vector<SettingSpec> _settingSpecs;
 
     std::shared_ptr<GFXBase> _GFX[NUM_CHANNELS];
 
+    template<typename Tv>
+    bool SetIfSelected(String settingName, const String& propertyName, Tv& property, const Tv& value)

```

Comment:
Does settingName need to be passed by value?

Reply:
Instinctively I'd say no, but I also find it difficult believe I randomly left one parameter unreferenced. I'll check and modify if appropriate.

Reply:
I checked, and I must not have had my morning coffee when I wrote that. There is no reason not to const reference that.
Good catch!